pandas/tests/generic/test_frame.py

```diff
@@ -269,3 +270,40 @@ def test_deepcopy_empty(self):
         empty_frame_copy = deepcopy(empty_frame)
 
         self._compare(empty_frame_copy, empty_frame)
+
+    @pytest.mark.skipif(os.name == 'nt',
+                        reason="Windows use \r\n for newline")
+    @pytest.mark.parametrize("header, index_label, expected", [
+        (False, True, 'index.name,,\n0,0,0\n1,0,0\n'),
+        (True, True, 'index.name,0,1\n0,0,0\n1,0,0\n'),
+        (False, False, '0,0,0\n1,0,0\n'),
+        (True, False, ',0,1\n0,0,0\n1,0,0\n'),
+        (False, None, '0,0,0\n1,0,0\n'),
+        (True, None, 'index.name,0,1\n0,0,0\n1,0,0\n')
+    ])
+    def test_to_csv_header_single_index(self, header, index_label, expected):
+        # issue 24546
+        df = pd.DataFrame(np.zeros((2, 2), dtype=int))
+        df.index.name = 'index.name'
```

Comment:
for added assurance it may be worth also assigning a name (different to the row index name) to the columns index in the test set-up

Reply:
added!

Reply:
i was thinking `df.columns.name` here.